apps/emqx/src/emqx_alarm.erl

```diff
@@ -113,6 +116,25 @@ activate(Name, Details) ->
 activate(Name, Details, Message) ->
     gen_server:call(?MODULE, {activate_alarm, Name, Details, Message}).
 
+-spec ensure_deactivated(binary() | atom()) -> ok.
+ensure_deactivated(Name) ->
+    ensure_deactivated(Name, no_details).
+
+-spec ensure_deactivated(binary() | atom(), map()) -> ok.
+ensure_deactivated(Name, Data) ->
+    ensure_deactivated(Name, Data, <<>>).
+
+ensure_deactivated(Name, Data, Message) ->
+    case mnesia:dirty_read(?ACTIVATED_ALARM, Name) of
```

Comment:
Looks like duplication of code:

```
handle_call({deactivate_alarm, Name, Details, Message}, _From, State) ->
    case mnesia:dirty_read(?ACTIVATED_ALARM, Name) of
        [] ->
            {reply, {error, not_found}, State};
        [Alarm] ->
            deactivate_alarm(Alarm, Details, Message),
            {reply, ok, State, get_validity_period()}
    end;
```

Is it an optimization to avoid unnecessary `gen_server` calls? In this case a comment would be nice.

Reply:
yes. added comments. squash pushed.